lib/IRGen/GenDecl.cpp

```diff
@@ -1659,10 +1659,14 @@ getIRLinkage(const UniversalLinkageInfo &info, SILLinkage linkage,
 
   case SILLinkage::HiddenExternal:
   case SILLinkage::PrivateExternal:
-    return std::make_tuple(isDefinition
-                               ? llvm::GlobalValue::AvailableExternallyLinkage
-                               : llvm::GlobalValue::ExternalLinkage,
-                           llvm::GlobalValue::HiddenVisibility,
+    if (isDefinition) {
+      return std::make_tuple(llvm::GlobalValue::AvailableExternallyLinkage,
```

Comment:
Why is the definition marked as `available_externally` if it is a hidden external?  Doesn't `PrivateLinkage` seem more appropriate?

Reply:
From what I understand, hidden_external gets used in SIL when a hidden function is de-serialized in a different module. We want the optimizer to be able to inline the function. The AvailableExternally linkage (which was there before my change) does that. For the case of a definition, my change preserves the previous behavior except for using DefaultStorageClass instead of DLLImport. It may well be the case that further changes are in order, but I would still suggest changing one thing at a time (especially to get the bots passing again).

Reply:
I don't think this can happen across modules anymore, but it can still show up in theory in multi-file builds (referencing something from another file in the same module). That wouldn't need any DLLImport or DLLExport, though.

Reply:
Ah, so, if it is for intra-module usage, then default DLL storage is correct.